### ISSUE TITLE
[ISSUE] longer job deployment with YAML configuration

### ISSUE THREAD

**Issue by michal-szklarczyk** [databrickslabs/dbx#111]
## Expected Behavior
Job deployment time using YAML configuration is similar to deployment time using JSON configuration 
## Current Behavior
- job deployment time with JSON configuration - 10sec
dbx deploy --jobs=b2s_dim_business_hubspot --environment=b2s_jobs_prod --deployment-file=conf/deployment_prod.json
[dbx][2021-12-10 09:07:23.217] Starting new deployment for environment b2s_jobs_prod
[dbx][2021-12-10 09:07:23.224] No environment variables provided, using the ~/.databrickscfg
[dbx][2021-12-10 09:07:24.931] Re-building package
[dbx][2021-12-10 09:07:26.264] Package re-build finished
[dbx][2021-12-10 09:07:26.264] Locating package file
[dbx][2021-12-10 09:07:26.277] Package file located in: dist\databricks_jobs-1639123645-py3-none-any.whl
[dbx][2021-12-10 09:07:26.291] Requirements file is not provided
[dbx][2021-12-10 09:07:26.291] Deployment will be performed only for the following jobs: ['b2s_dim_business_hubspot']
[dbx][2021-12-10 09:07:27.361] Deploying file: dist\databricks_jobs-1639123645-py3-none-any.whl
[dbx][2021-12-10 09:07:29.262] Deploying file: jobs\batch\b2s\dim_business_hubspot.py
[dbx][2021-12-10 09:07:30.098] Updating job definitions
[dbx][2021-12-10 09:07:30.099] Processing deployment for job: b2s_dim_business_hubspot
[dbx][2021-12-10 09:07:30.328] Updating existing job with id: 5093 and name: b2s_dim_business_hubspot
[dbx][2021-12-10 09:07:31.960] Updating job definitions - done
[dbx][2021-12-10 09:07:33.209] Deployment for environment b2s_jobs_prod finished successfully

- job deployment time with YAML configuration - 14min
dbx deploy --jobs=b2s_dim_business_hubspot --environment=b2s_jobs_prod --deployment-file=conf/deployment_prod.yaml
[dbx][2021-12-10 11:36:21.829] Starting new deployment for environment b2s_jobs_prod
[dbx][2021-12-10 11:36:21.836] No environment variables provided, using the ~/.databrickscfg
[dbx][2021-12-10 11:36:23.423] Re-building package
[dbx][2021-12-10 11:36:24.635] Package re-build finished
[dbx][2021-12-10 11:36:24.635] Locating package file
[dbx][2021-12-10 11:36:24.646] Package file located in: dist\databricks_jobs-1639132584-py3-none-any.whl
[dbx][2021-12-10 11:36:24.672] Requirements file is not provided
[dbx][2021-12-10 11:36:24.673] Deployment will be performed only for the following jobs: ['b2s_dim_business_hubspot']
[dbx][2021-12-10 11:36:25.540] Deploying file: .
[dbx][2021-12-10 11:50:50.117] File is already stored in the deployment, no action needed
[dbx][2021-12-10 11:50:50.347] Deploying file: dist\databricks_jobs-1639132584-py3-none-any.whl
[dbx][2021-12-10 11:50:52.150] Deploying file: jobs\batch\b2s\dim_business_hubspot.py
[dbx][2021-12-10 11:50:52.825] Updating job definitions
[dbx][2021-12-10 11:50:52.825] Processing deployment for job: b2s_dim_business_hubspot
[dbx][2021-12-10 11:50:53.052] Updating existing job with id: 5093 and name: b2s_dim_business_hubspot
[dbx][2021-12-10 11:50:54.667] Updating job definitions - done
[dbx][2021-12-10 11:50:55.936] Deployment for environment b2s_jobs_prod finished successfully
## Steps to Reproduce (for bugs)
Configure a simple task and attempt to deploy to an existing cluster using dbx deploy with JSON and YAML configuration.
Below is a comparison of both configurations 
![obraz](https://user-images.githubusercontent.com/91941928/145603134-38ac0aa7-35af-48d6-97ff-e4e587727a68.png)

## Your Environment

* dbx version used: 0.2.0 and 0.2.2
* Databricks Runtime version: 9.1 LTS

**Comment by renardeinside** [comment]
hi @michal-szklarczyk , 

I see the following lines in the log you've provided:
```
[dbx][2021-12-10 11:36:25.540] Deploying file: .
[dbx][2021-12-10 11:50:50.117] File is already stored in the deployment, no action needed
```

It seems to me you need to enable the path adjustment logic and adopt your YAML as described [here](https://dbx.readthedocs.io/en/latest/path_adjustment.html).

**Comment by michal-szklarczyk** [comment]
Hi @renardeinside,
Thank you for your fast response.
My problem was related to evaluating empty string in email_notification section:
Bad solution:
![obraz](https://user-images.githubusercontent.com/91941928/145832245-4d07c0c2-80a8-4279-a762-1930f803071f.png)

Good:
![obraz](https://user-images.githubusercontent.com/91941928/145834534-3c36f4ac-6839-4bd8-a8cc-8b1f390ed4c8.png)


I also noticed some problems with feeding library to install on the cluster.
The method shown in https://dbx.readthedocs.io/en/latest/quickstart.html did not work for me

![obraz](https://user-images.githubusercontent.com/91941928/145834194-3d6e1a43-57c8-4ab9-aecc-6fed6c8261a0.png)

**Comment by renardeinside** [comment]
it shall look like this - I'll update the doc:
```
libraries:
  - pypi: 
     package: pymongo
```
Thank you for pointing this out @michal-szklarczyk

**Comment by renardeinside** [comment]
documentation issue is addressed in #113 
thanks a lot for pointing this out @michal-szklarczyk 👍
